### PR TITLE
Basic conditionals

### LINKED ISSUE #50: Basic conditionals
```
<if condition="${someBool}">
  <then>
    <p>It's true!</p>
  </then>
  <else>
    <p>It's false!</p>
  </else>
</if>
```

This could also be a way to test if a fill exists before using it, if `""` is "falsey". Maybe `exists` could be an alternative to the `condition` attribute that works like that, instead of conflating with bools:

```
<if exists="${dubious}">
  <then>
    <p>Hello, ${dubious}!</p>
  </then>
</if>
```

could the "then" be optional?

```
<if exists="${dubious}">
    <p>Hello, ${dubious}!</p>
</if>
```

Pretty easy to implement, question is whether it's a good idea or pointless complexity. :)

### PR DESCRIPTION
**Pull request by emhoracek**
To fix #50

### CODE REVIEW

File: test/Spec.hs [hunk 1]
```diff
@@ -450,6 +452,89 @@ statefulTests =
...
+      it "should display the stuff within the `then` tag" $ do
+        hLarcenyState.lSubs .=
+           subs [("if", ifFill)
+                ,("existing", textFill "exists")]
+        template `shouldRenderM` "It exists!"
+    describe "the fill doesn't exist" $ do
+      it "should display the stuff within the `else` tag" $ do
+        hLarcenyState.lSubs .=
+           subs [("if", ifFill)]
+        template `shouldRenderM` "It doesn't exist!"
```
Comment (dhartunian) [inline test/Spec.hs]: also test "False" and "" fills for `existing`

File: test/Spec.hs [hunk 2]
```diff
@@ -450,6 +452,89 @@ statefulTests =
...
+      it "should display the stuff within the `then` tag" $ do
+        hLarcenyState.lSubs .=
+           subs [("if", ifFill)
+                ,("existing", textFill "exists")]
+        template "True" `shouldRenderM` "It exists!"
+    describe "any other combination" $ do
+      it "should display the stuff within the `else` tag" $ do
+        hLarcenyState.lSubs .=
+           subs [("if", ifFill)]
+        template"True" `shouldRenderM` "It doesn't exist and/or it's false!"
```
Comment (dhartunian) [inline test/Spec.hs]: missing space here

File: test/Spec.hs (hunk 1 above)
Comment (emhoracek) [reply test/Spec.hs]: ummmmm how do you feel about the specified behavior of "exists" being that if it's a blank string then it doesn't exist? test:

```(Haskell)
describe "the fill is an empty string" $
      it "should display the stuff within the `else` tag" $ do
        hLarcenyState.lSubs .=
           subs [("if", ifFill)
                ,("existing", textFill "")]
        template `shouldRenderM` "It doesn't exist!"
```

Comment (emhoracek) [reply test/Spec.hs]: as a first approximation I mean

Comment (dhartunian) [reply test/Spec.hs]: 👍

**Pull request merged**